cpp/arcticdb/python/normalization_checks.cpp

```diff
@@ -122,17 +122,15 @@ void update_rowcount_normalization_data(
             );
 
             size_t new_start = new_index->start();
-            if (new_start != 0) {
-                auto stop = old_index->start() + old_length * old_index->step();
-                normalization::check<ErrorCode::E_INCOMPATIBLE_INDEX>(
-                    new_start == stop,
-                    "The appending data has a RangeIndex.start={} that is not contiguous with the {}"
-                    "stop ({}) of",
-                    error_suffix,
-                    new_start,
-                    stop
-                );
-            }
+            auto stop = old_index->start() + old_length * old_index->step();
+            normalization::check<ErrorCode::E_INCOMPATIBLE_INDEX>(
+                new_start == stop || (new_start == 0 && new_index->step() == 1),
```

Comment:
is this going to allow something like this to happen
     a
0   0
1   1
2   2

Append index (0, 1, 2)

     a
0   0
1   1
2   2
0   3
1   4
2   5

Or is there some other check preventing it?

Reply:
We have to allow appending index `[0, 1, 2` to index `[0, 1, 2]`, otherwise our default behaviour when you don't provide an index won't work. We only store the start and step though, so it will come out as:
```
0 0
1 1
2 2
3 3
4 4
5 5
```